.github/workflows/build.yaml

```diff
@@ -1,15 +1,14 @@
 name: build
 
 on:
+  workflow_dispatch:
+  pull_request:
   push:
     branches:
-      - '**'
-  pull_request:
-    paths:
-      - '**'
+    - main
 
 jobs:
-  build:
+  binaries:
```

Comment:
This can be tackled as a follow-up, and this is only an issue in the short term until e2e is fleshed out, but it looks like we have a job lower in this build workflow for ensuring the root Dockerfile container image can be successfully built. We should also verify that built image can be run, even if it's just running the `--help` flag for the registry+v1 or k8s provisioners.